Plugins: Allow overriding Prometheus URL in query editor link

tbregolin opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/master/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the master branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**:
This PR allows for the "Link to Graph in Prometheus" in the query editor to use a different URL than the Prometheus datasource itself.

**Which issue(s) this PR fixes**:
When using a Prometheus datasource in "Server" (proxied) mode, it may be
desirable to have different URLs for the datasource itself used by
Grafana and the "Link to Graph in Prometheus" button in the query editor
UI. That is useful, in particular, when the Prometheus endpoints are under
different forms of authentication for service users (e.g. Grafana) and
other user-agents (e.g. a person with a browser).
<!--

* Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

**Special notes for your reviewer**:

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=23284) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=23284) before we can accept your contribution.<br/><hr/>**Thomas Bregolin** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=23284) it.</sub>

tbregolin committed:
Allow overriding Prometheus URL in query editor (#23287)

When using a Prometheus datasource in "Server" (proxied) mode, it may be
desirable to have different URLs for the datasource itself used by
Grafana and the "Link to Graph in Prometheus" button in the query editor
UI. That is useful, in particular, when the Prometheus UI is under
different forms of authentication for service users (e.g. Grafana) and
other user-agents (e.g. a person with a browser).

torkelo commented:
Thanks for contributing! 

Can you open a feature request so we can evaluate how wide spread this use case is? We try to limit UI options as much as possible to not overwhelm the user with options for every edge case, so getting a better sense of need for this is required before merging

tbregolin commented:
> Thanks for contributing!
> 
> Can you open a feature request so we can evaluate how wide spread this use case is? We try to limit UI options as much as possible to not overwhelm the user with options for every edge case, so getting a better sense of need for this is required before merging

Thank you for reviewing.

I have opened the feature request here: https://github.com/grafana/grafana/issues/23287

roman-vynar commented:
This would be very helpful! 
Our datasource is http://localhost:9090 (server method) and I always post-edit the link in the browser address bar after clicking it in Grafana.
